rust/candid/src/types/impls.rs

```diff
@@ -1,16 +1,159 @@
 use super::internal::*;
-use super::{CandidType, Compound, Serializer};
+use super::{CandidType, CandidTypeCache, CandidTyping, Compound, IdlSerialize, Serializer};
+
+macro_rules! serialize_impl {
```

Comment:
Too many macros affects readability. Although the code is shorter, it is harder to debug.